### PR TITLE
Resource Library 

### PR DESCRIPTION
**Pull request by manojSRawat**

### CODE REVIEW

File: projects/resource-library/src/lib/components/library-filter/library-filter.component.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  public telemetryPageId: string;
+  private onComponentDestroy$ = new Subject<any>();
+  public frameworkDetails: any = {};
+  public currentFilters: any;
+  public searchQuery: string;
+
+  constructor(private frameworkService: FrameworkService,
+              public editorService: EditorService,
+              public telemetryService: EditorTelemetryService,
+              public treeService: TreeService,
```
Comment (pallakartheekreddy) [inline projects/resource-library/src/lib/components/library-filter/library-filter.component.ts:37]: is `treeService` being used?

File: projects/resource-library/src/lib/components/library-filter/library-filter.component.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      contentTypes = _.map(this.helperService.contentPrimaryCategories, 'name');
+    }*/
+    const contentTypes = _.map(this.helperService.contentPrimaryCategories, 'name');
+
+    this.currentFilters = {
+      primaryCategory: contentTypes,
+      board: [],
+      medium: [],
+      gradeLevel: [],
+      subject: [],
```
Comment (pallakartheekreddy) [inline projects/resource-library/src/lib/components/library-filter/library-filter.component.ts:59]: Can we avoid hardcoding framework categories?

File: projects/resource-library/src/lib/components/contentplayer-page/contentplayer-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    setTimeout(() => {
+      const playerElement = this.contentIframe.nativeElement;
+      playerElement.src = iFrameSrc;
+      playerElement.onload = (event) => {
+        try {
+          this.adjustPlayerHeight();
+          // this.playerLoaded = true;
+          playerElement.contentWindow.initializePreview(this.playerConfig);
+          // playerElement.addEventListener('renderer:telemetry:event', telemetryEvent => this.generateContentReadEvent(telemetryEvent));
+          // window.frames['contentPlayer'].addEventListener('message', accessEvent => this.generateScoreSubmitEvent(accessEvent), false);
```
Comment (pallakartheekreddy) [inline projects/resource-library/src/lib/components/contentplayer-page/contentplayer-page.component.ts:70]: remove un used code

File: projects/resource-library/src/lib/components/contentplayer-page/contentplayer-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        } catch (err) {
+          console.log('loading default player failed', err);
+          // const prevUrls = this.navigationHelperService.history;
+          // if (this.isCdnWorking.toLowerCase() === 'yes' && prevUrls[prevUrls.length - 2]) {
+          //   history.back();
+          // }
```
Comment (pallakartheekreddy) [inline projects/resource-library/src/lib/components/contentplayer-page/contentplayer-page.component.ts:76]: remove un used code

File: projects/resource-library/src/lib/components/library-player/library-player.component.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+export class LibraryPlayerComponent implements OnInit {
+@Input() contentListDetails;
+@Output() moveEvent = new EventEmitter<any>();
+  constructor(public telemetryService: EditorTelemetryService, public editorService: EditorService,
+              public configService: ConfigService) { }
+
+  ngOnInit() {
+  }
+  addToLibrary() {
+    if (this.editorService.checkIfContentsCanbeAdded()) {
```
Comment (pallakartheekreddy) [inline projects/resource-library/src/lib/components/library-player/library-player.component.ts:19]: As per the design, `openHierarchyPopup` should be configurable. How are we handling it?

File: projects/resource-library/src/lib/components/library/library.component.ts [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+  public isFilterOpen = false;
+  collectionhierarcyData: any;
+  public defaultFilters: any;
+  pageStartTime: any;
+  selectedUnit: string;
+  public frameworkId: any;
+  constructor(public telemetryService: EditorTelemetryService,
+              private editorService: EditorService,
+              private router: Router,
+              private treeService: TreeService,
```
Comment (pallakartheekreddy) [inline projects/resource-library/src/lib/components/library/library.component.ts:41]: if  `treeService` not being used anywhere. let's remove it.

File: projects/resource-library/src/lib/components/library/library.component.ts [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+  ngOnInit() {
+    this.frameworkService.initialize(_.get(this.libraryInput, 'framework'));
+    this.editorService.initialize(_.get(this.libraryInput, 'editorConfig'));
+    this.treeService.initialize(_.get(this.libraryInput, 'editorConfig'));
+    this.telemetryService.initializeTelemetry(_.get(this.libraryInput, 'editorConfig'));
+    // const activeNode = this.treeService.getActiveNode();
+    // this.selectedUnit = _.get(activeNode, 'data.id');
+    this.collectionId = _.get(this.libraryInput, 'collectionId');
+    this.searchFormConfig = _.get(this.libraryInput, 'searchFormConfig');
+    this.editorService.fetchCollectionHierarchy(this.collectionId).subscribe((response: any) => {
```
Comment (pallakartheekreddy) [inline projects/resource-library/src/lib/components/library/library.component.ts:58]: are we expecting `collectionId` to be available every time?

File: projects/resource-library/src/lib/components/library/library.component.ts [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+
+    if (_.isEmpty(contentTypes)) {
+      contentTypes = _.map(this.helperService.contentPrimaryCategories, 'name');
+    }*/
+    this.defaultFilters = _.pickBy({
+      primaryCategory: [_.get(this.collectionhierarcyData, 'primaryCategory')] || _.get(this.collectionhierarcyData, 'primaryCategories'),
+      board: [_.get(this.collectionhierarcyData, 'board')] || _.get(this.collectionhierarcyData, 'boardIds'),
+      gradeLevel: _.get(this.collectionhierarcyData, 'gradeLevel') || _.get(this.collectionhierarcyData, 'gradeLevelIds'),
+      medium: _.get(this.collectionhierarcyData, 'medium') || _.get(this.collectionhierarcyData, 'mediumIds'),
+      subject: _.get(this.collectionhierarcyData, 'subject') || _.get(this.collectionhierarcyData, 'subjectIds'),
```
Comment (pallakartheekreddy) [inline projects/resource-library/src/lib/components/library/library.component.ts:113]: let's not try hardcoding framework categories

File: projects/resource-library/src/lib/components/resource-reorder/resource-reorder.component.ts [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+import { ToasterService } from '../../services/toaster/toaster.service';
+
+
+@Component({
+  selector: 'lib-resource-reorder',
+  templateUrl: './resource-reorder.component.html',
+  styleUrls: ['./resource-reorder.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class ResourceReorderComponent implements OnInit {
```
Comment (pallakartheekreddy) [inline projects/resource-library/src/lib/components/resource-reorder/resource-reorder.component.ts:15]: Is this component used in any of the flows?

**Review by pallakartheekreddy — CHANGES_REQUESTED** [review]
@bharatkashyap @aayushkedia95  There are a lot of files which is not relevant. let's try to identify and remove from the codebase.
ex: Mathmodal, latexServcie, Config files.

File: projects/resource-library/src/lib/components/library-filter/library-filter.component.ts (hunk 1 above)
Comment (manojSRawat) [reply projects/resource-library/src/lib/components/library-filter/library-filter.component.ts]: No, removed in the latest commit.

File: projects/resource-library/src/lib/components/library-filter/library-filter.component.ts (hunk 2 above)
Comment (manojSRawat) [reply projects/resource-library/src/lib/components/library-filter/library-filter.component.ts]: fixed

File: projects/resource-library/src/lib/components/contentplayer-page/contentplayer-page.component.ts (hunk 4 above)
Comment (manojSRawat) [reply projects/resource-library/src/lib/components/contentplayer-page/contentplayer-page.component.ts]: removed in the latest commit.

File: projects/resource-library/src/lib/components/library-player/library-player.component.ts (hunk 5 above)
Comment (manojSRawat) [reply projects/resource-library/src/lib/components/library-player/library-player.component.ts]: we're working on this.

File: projects/resource-library/src/lib/components/resource-reorder/resource-reorder.component.ts (hunk 9 above)
Comment (manojSRawat) [reply projects/resource-library/src/lib/components/resource-reorder/resource-reorder.component.ts:15]: currently, it's used in the library component. Will remove if not required.

**Review by pallakartheekreddy — APPROVED** [review]

**Pull request merged**